crates/core/tedge/src/cli/connect/command.rs

```diff
@@ -274,7 +270,11 @@ impl ConnectCommand {
             Cloud::Azure(_) => (),
         }
 
-        Ok(())
+        if connection_check_success {
+            Ok(())
+        } else {
+            Err(anyhow::anyhow!("Connection check failed").into())
+        }
```

Comment:
note: To make the `tedge connect aws` return a non-zero exit code when the connection check failed had to add this here, which can influence `c8y` where failing the connection check is apparently not a failure? I'm not sure why, and if it's intended or accidental.

Reply:
Yeah I remember the decision.
At the time, the purpose of `tedge connect` was to establish the bridge to the cloud, and the connection check was considered an optional check because it may fail, for example, when the server is not responded, that's not thin-edge side of failure. That's why we decided to have exit code `0`.

However, now we have `--offline` option. So, I don't mind if we return non-zero exist code when connection check fails.